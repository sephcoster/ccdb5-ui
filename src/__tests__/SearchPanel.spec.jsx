import { SearchPanel } from '../SearchPanel'
import React from 'react'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'

function setupSnapshot(initialStore={}) {
  const results = Object.assign({}, initialStore)

  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: {},
    results
  })

  return renderer.create(
    <Provider store={ store } >
      <IntlProvider locale="en">
        <SearchPanel lastIndexed={ results.lastIndexed } />
      </IntlProvider>
    </Provider>
  )
}

describe('component:SearchPanel', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot()
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays last updated date when present', () => {
    const target = setupSnapshot({ lastIndexed: new Date( '2016-02-01T05:00:00.000Z' ) })
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
