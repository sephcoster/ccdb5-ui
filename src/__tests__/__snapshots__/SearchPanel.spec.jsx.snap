// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:SearchPanel displays last updated date when present 1`] = `
<div
  className="search-panel"
>
  <h2>
    Search complaint data 
    <span
      className="date-subscript"
    >
      (last updated: 
      <span>
        2/1/2016
      </span>
      )
    </span>
  </h2>
  <div>
    <nav
      className="search-bar"
    >
      <form
        action=""
        onSubmit={[Function]}
      >
        <h4>
          Search Within
        </h4>
        <div
          className="layout-row"
        >
          <div
            className="cf-select flex-fixed"
          >
            <select
              aria-label="Choose which field will be searched"
              id="searchField"
              onChange={[Function]}
              value={undefined}
            >
              <optgroup
                label="Search Within"
              >
                <option
                  value="all"
                >
                  All Data
                </option>
                <option
                  value="company"
                >
                  Company Name
                </option>
                <option
                  value="complaint_what_happened"
                >
                  Narratives
                </option>
              </optgroup>
            </select>
          </div>
          <div
            className="flex-all typeahead-portal"
          >
            <section
              className="typeahead "
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                className="m-btn-inside-input input-contains-label"
              >
                <label
                  className="input-contains-label_before input-contains-label_before__search"
                />
                <input
                  aria-label="The term to search for"
                  autoComplete="off"
                  className="a-text-input"
                  id="searchText"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Enter your search term(s)"
                  type="text"
                  value=""
                />
              </div>
            </section>
          </div>
          <button
            className="a-btn flex-fixed"
            type="submit"
          >
            Search
          </button>
          <div
            className="advanced-container flex-fixed"
          >
            <a
              className="a-btn a-btn__link o-expandable_cue-open"
              onClick={[Function]}
            >
              Show advanced search tips
            </a>
          </div>
        </div>
      </form>
    </nav>
  </div>
</div>
`;

exports[`component:SearchPanel renders without crashing 1`] = `
<div
  className="search-panel"
>
  <h2>
    Search complaint data 
  </h2>
  <div>
    <nav
      className="search-bar"
    >
      <form
        action=""
        onSubmit={[Function]}
      >
        <h4>
          Search Within
        </h4>
        <div
          className="layout-row"
        >
          <div
            className="cf-select flex-fixed"
          >
            <select
              aria-label="Choose which field will be searched"
              id="searchField"
              onChange={[Function]}
              value={undefined}
            >
              <optgroup
                label="Search Within"
              >
                <option
                  value="all"
                >
                  All Data
                </option>
                <option
                  value="company"
                >
                  Company Name
                </option>
                <option
                  value="complaint_what_happened"
                >
                  Narratives
                </option>
              </optgroup>
            </select>
          </div>
          <div
            className="flex-all typeahead-portal"
          >
            <section
              className="typeahead "
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                className="m-btn-inside-input input-contains-label"
              >
                <label
                  className="input-contains-label_before input-contains-label_before__search"
                />
                <input
                  aria-label="The term to search for"
                  autoComplete="off"
                  className="a-text-input"
                  id="searchText"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Enter your search term(s)"
                  type="text"
                  value=""
                />
              </div>
            </section>
          </div>
          <button
            className="a-btn flex-fixed"
            type="submit"
          >
            Search
          </button>
          <div
            className="advanced-container flex-fixed"
          >
            <a
              className="a-btn a-btn__link o-expandable_cue-open"
              onClick={[Function]}
            >
              Show advanced search tips
            </a>
          </div>
        </div>
      </form>
    </nav>
  </div>
</div>
`;
